### ISSUE TITLE
Web - IOU - Currency is changed to location selected right after changing the country

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#10022]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
Issue found when executing #9840
## Action Performed:
1. Open some chat report and Request money modal
2. Make sure that the currency is CA$
3. Now, change your location to some country using EUR (Netherland)

## Expected Result:
Open the request money modal again, the currency should be CA$ still as we have just fetched the new currency on opening the modal

## Actual Result:
Currency is updated to "Euro"

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:**  v1.1.85 -5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**Email or phone of affected tester (no customers):** [kavi.utest+chatd@gmail.com](mailto:kavi.utest+chatd@gmail.com) / Test1234
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/179882142-eb3217bc-720c-4914-8280-c5610fa2065b.mp4

**Expensify/Expensify Issue URL:** 
**Issue reported by:** Applause internal testers
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by yuwenmemon** [comment]
Hmmm... I'm actually confused as to why we think this is a bug? Adding a contributor manager to give a second opinion

**Comment by MitchExpensify** [comment]
[Upwork Job](https://www.upwork.com/jobs/~0135538fe076b09dcf)

**Comment by wildan-m** [comment]
### Proposal

`componentDidMount` will always be called when the modal is re-opened, to make sure the currency option persist, we need to call ` IOU.setIOUSelectedCurrency` function if there is no value in `this.props.iou.selectedCurrencyCode`

```diff
diff --git a/src/pages/iou/IOUModal.js b/src/pages/iou/IOUModal.js
index 14a24b473..80fba9a29 100755
--- a/src/pages/iou/IOUModal.js
+++ b/src/pages/iou/IOUModal.js
@@ -145,7 +145,9 @@ class IOUModal extends Component {
 
     componentDidMount() {
         PersonalDetails.openIOUModalPage();
-        IOU.setIOUSelectedCurrency(this.props.currentUserPersonalDetails.localCurrencyCode);
+        
+        if(!this.props.iou.selectedCurrencyCode)
+            IOU.setIOUSelectedCurrency(this.props.currentUserPersonalDetails.localCurrencyCode);
     }
 
     componentDidUpdate(prevProps) {
111

**Comment by Santhosh-Sellavel** [comment]
@puneetlath or @MitchExpensify  Is this a valid bug?

**Comment by MitchExpensify** [comment]
Yes on second look I do not think this is a bug. Will hold on @puneetlath 's take before possibly removing the Upwork job and closing this out

**Comment by puneetlath** [comment]
Hm, yes I agree this is the expected behavior and not a bug. Closing!

**Comment by MitchExpensify** [comment]
Upwork Job removed
